pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -204,10 +202,15 @@ func (nm *NodeManager) GetNode(nodeName k8stypes.NodeName) (v1.Node, error) {
 	return *node, nil
 }
 
-func (nm *NodeManager) addNode(node *v1.Node) {
+func (nm *NodeManager) addNode(node *v1.Node) bool {
 	nm.registeredNodesLock.Lock()
+	if _, ok := nm.registeredNodes[node.ObjectMeta.Name]; !ok {
```

Comment:
I did not get this logic. If the node is not already registered, dont we have to register it? We seem to be just returning.

I liked the earlier logic of always adding the node to the map with the updated Node object irrespective of whether it is already in the map or not. This way the discovery process can consider the new Node details while searching for the node VM in the pool of VCs. What is the reason for changing this logic?

Reply:
Actually we don't need this change at all. Let me remove this change.